dark/fasta.py

```diff
@@ -94,15 +94,17 @@ def iter(self):
         instance of the desired read class.
         """
         for seq in SeqIO.parse(self.file_, 'fasta'):
-            yield self.readClass(seq.description, str(seq.seq))
+            read = self.readClass(seq.description, str(seq.seq))
+            if read.checkAlphabeth():
```

Comment:
I don't think you need the `if` here. Just call `checkAlphabet` and let it raise if there's a problem.